Documentation [16503] (#64)
eProsima/AML-IP: amlip_docs/rst/developer_manual/installation/sources/linux/linux.rst

.. _gtest_installation:

Gtest
^^^^^

Gtest_ is a unit testing library for C++.
For a detailed description of the Gtest_ installation process, please refer to the
`Gtest Installation Guide <https://github.com/google/googletest>`_.

If using colcon_, it is also possible to clone the Gtest_ Github repository into the |ddsrouter|
workspace and compile it as a dependency package.
Add this new package to :code:`.repos` file before importing the sources:

.. code-block:: bash

    googletest-distribution:
        type: git
        url: https://github.com/google/googletest.git
        version: release-1.12.1

.. _eprosima_dependencies:

eProsima dependencies
^^^^^^^^^^^^^^^^^^^^^

These are the eProsima libraries required for building |amlip|:

- ``foonathan_memory_vendor``, an STL compatible C++ memory allocation library.
- ``fastcdr``, a C++ library that serializes according to the standard CDR serialization mechanism.
- ``fastrtps``, the core library of eProsima Fast DDS library.
- ``cmake_utils``, an eProsima utilities library for CMake.
- ``cpp_utils``, an eProsima utilities library for C++.
- ``ddsrouter_core``, the eProsima DDS Router library C++.

> If it already exists in the system an installation of these libraries there is no need to build them again,

Reviewer: ```suggestion
If they are already available in the system there is no need to build them again,
```

just source them when building the |amlip|.
If using CMake_, add the already libraries installation paths to :code:`LD_LIBRARY_PATH`.
If using colcon_, use the following command to source them:

.. code-block:: bash

    source <eprosima-dependencies-installation-path>/install/setup.bash

Installation methods
====================

There are two main possibilities to build |amlip| from sources in Linux.
One of them uses CMake_ and the other colcon_, an auto-build framework.

> **colcon_ version is advice for non advance users as it is easier and neater.**

Reviewer 2: Change also in Windows.

.. toctree::
    :maxdepth: 1

>     colcon <linux_colcon>

Reviewer 2: Change in Windows
